src/test/ruby/ssl/test_socket.rb

```diff
@@ -136,6 +136,21 @@ def test_connect_nonblock
     end
   end if RUBY_VERSION > '2.2'
 
+  def test_minmax_ssl_version
+    # WIP: this test should setup it's own server to prove that SSLContext.min_version and SSLContext.max_version work as expected
+    # However below line is failing  with "no cipher match" even on master
+    # When "no cipher match" is fixed, then this test need cleanup
+
+    #ssl_server = server         
+    begin
+      ssl_client = client(443, host: "google.com", min_version: OpenSSL::SSL::TLS1_VERSION, max_version: OpenSSL::SSL::TLS1_1_VERSION, ciphers: nil)
```

Comment:
believe we want some unit tests (done above except for the `set_minmax_proto_version`) for the version parsing/mapping instead of an integration tests here. 
also not sure hard-coding what google has on their servers atm. 
anyhow it's a good start but we might want to do an integration tests for this [elsewhere](https://github.com/jruby/jruby-openssl/tree/master/src/test/integration).

Reply:
Yes, totally agree.

This PR will still need to be polished, more tests added.
But first I believe that master branch should be stabilized.

Reply:
Ok, I've added integration test in `src/test/integration/ssl_test.rb` like you suggested.
`test_socket.rb` still needs improvement.

Reply:
should we remove this than, it's very dependend on what google.com has set on its servers, right?

Reply:
Yes, let me remove it later since I've added the integration test. 
What do you think about the integration test then?

Later I will try to rework test in `test_socket.rb` or remove completely since it cannot be tested now due to another issue (`server` method fails with `no ciphers match`) at least on my env.

Reply:
> Yes, let me remove it later since I've added the integration test.

sorry but I would prefer not to deal with leftovers as they will bite someone later. 
I am fully aware CI testing is broken atm but that is no reason to leave externally dependent tests here.
you can keep it around for yourself in a follow-up branch.

> What do you think about the integration test then?

we do not run those atm regularly, so I am fine with having them as you did. thanks.

Reply:
Done. I Removed this test.
Instead got inspired by `test_ssl.rb` and implemented parameterized test for this issue.